tests/components/androidtv/test_config_flow.py

```diff
@@ -0,0 +1,446 @@
+
+
+async def test_abort_if_host_exist(hass):
+    """Test we abort if component is already setup."""
+    MockConfigEntry(
+        domain=DOMAIN, data=CONFIG_ADB_SERVER, unique_id=ETH_MAC
+    ).add_to_hass(hass)
+
+    config_data = CONFIG_ADB_SERVER.copy()
+    config_data[CONF_HOST] = "name"
+    # Should fail, same IP Address (by PATCH_GET_HOST_IP)
+    with PATCH_GET_HOST_IP:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=config_data,
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_abort_import_if_host_exist(hass):
+    """Test we abort if component is already setup."""
+    MockConfigEntry(
+        domain=DOMAIN, data=CONFIG_ADB_SERVER, unique_id=ETH_MAC
+    ).add_to_hass(hass)
+
+    # Should fail, same Host in entry
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=CONFIG_ADB_SERVER,
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_abort_if_unique_exist(hass):
+    """Test we abort if component is already setup."""
+    config_data = CONFIG_ADB_SERVER.copy()
+    config_data[CONF_HOST] = "127.0.0.2"
+    MockConfigEntry(domain=DOMAIN, data=config_data, unique_id=ETH_MAC).add_to_hass(
+        hass
+    )
+
+    # Should fail, same SerialNo
+    with patch(
+        CONNECT_METHOD,
+        return_value=MockConfigDevice(),
+    ), PATCH_GET_HOST_IP:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_USER},
+            data=CONFIG_ADB_SERVER,
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "already_configured"
+
+
+async def test_on_connect_failed(hass):
+    """Test when we have errors connecting the router."""
+    flow_result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER, "show_advanced_options": True},
+    )
+
+    with patch(
+        CONNECT_METHOD,
+        return_value=None,
+    ), PATCH_GET_HOST_IP:
+        result = await hass.config_entries.flow.async_configure(
+            flow_result["flow_id"], user_input=CONFIG_ADB_SERVER
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "cannot_connect"}
+
+    with patch(
+        CONNECT_METHOD,
+        side_effect=TypeError,
+    ), PATCH_GET_HOST_IP:
+        result = await hass.config_entries.flow.async_configure(
+            flow_result["flow_id"], user_input=CONFIG_ADB_SERVER
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["errors"] == {"base": "unknown"}
```

Comment:
Follow the config flow all the way through to make sure it can recover from the error and still setup